shelley/chain-and-ledger/executable-spec/test/Test/Shelley/Spec/Ledger/Utils.hs

```diff
@@ -35,37 +40,52 @@ import           Hedgehog (MonadTest, (===))
 import           Shelley.Spec.Ledger.BaseTypes (Globals (..), ShelleyBase, UnitInterval,
                      mkActiveSlotCoeff, mkUnitInterval)
 import           Shelley.Spec.Ledger.Coin (Coin (..))
-import           Shelley.Spec.Ledger.Keys (pattern SKey, pattern SKeyES, pattern VKey,
-                     pattern VKeyES, pattern VKeyGenesis, hashKey, updateKESKey, vKey)
+import           Shelley.Spec.Ledger.Keys (KeyRole (..), hashKey, updateKES, vKey)
 import           Shelley.Spec.Ledger.OCert (KESPeriod (..))
 import           Shelley.Spec.Ledger.Slot (EpochNo, EpochSize (..), SlotNo)
 import           Shelley.Spec.Ledger.TxData (pattern Addr, pattern KeyHashObj, pattern StakeRefBase)
 
 import           Test.Cardano.Crypto.VRF.Fake (WithResult (..))
-import           Test.Shelley.Spec.Ledger.ConcreteCryptoTypes (Addr, CertifiedVRF, KeyPair, SKey,
-                     SKeyES, SignKeyVRF, VKey, VKeyES, VKeyGenesis, VerKeyVRF)
+import           Test.Shelley.Spec.Ledger.ConcreteCryptoTypes (Addr, CertifiedVRF, KeyPair,
+                     SignKeyDSIGN, SignKeyKES, SignKeyVRF, VKeyGenesis, VerKeyKES,
+                     VerKeyVRF, VKey, pattern VKey)
 
 assertAll :: (MonadTest m, Show a, Eq a) => (a -> Bool) -> [a] -> m ()
 assertAll p xs = [] === filter (not . p) xs
 
+-- | Construct a seed from a bunch of Word64s
+--
+--   We multiply these words by some extra stuff to make sure they contain
+--   enough bits for our seed.
+mkSeedFromWords
+  :: (Word64, Word64, Word64, Word64, Word64)
+  -> Seed
+mkSeedFromWords (w1, w2, w3, w4, w5) = mkSeedFromBytes . BSL.toStrict $
+  toByteString (w1 * 15485867) <>
+  toByteString (w2 * 32452867) <>
+  toByteString (w3 * 49979693) <>
+  toByteString (w4 * 67867979) <>
+  toByteString (w5 * 86028157) <>
+  toByteString (2147483647 :: Word64)
```

Comment:
The `toByteString` method on `Word64`s produces a decimal encoding using ascii digits.
see [here](https://github.com/twittner/bytestring-conversion/blob/master/src/Data/ByteString/Conversion/To.hs#L69) and [here](https://hackage.haskell.org/package/bytestring-0.10.10.0/docs/Data-ByteString-Builder.html#v:word64Dec). I suspect that's not what you wanted.

If you want a packed encoding, you can use [this](https://hackage.haskell.org/package/bytestring-0.10.10.0/docs/Data-ByteString-Builder.html#v:word64LE).

Reply:
Actually, this was quite deliberate. Using ASCII decimal digits packs up the representation more, so I get a bigger seed for my words. Of course, then I had to multiply by some big numbers _anyway_ because I still didn't have enough, so a packed representation would also work now. But there's no harm in using the decimal digits still, this is only being used for (pseudo-) randomness.